security/azure/sql_ingress_no_inbound_any/sql_no_ingress_any.pt

```diff
@@ -0,0 +1,192 @@
+    header "User-Agent", "RS Policies"
+  end
+
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "id", jmes_path(col_item, "id")
+      field "name", jmes_path(col_item, "name")
+      field "location", jmes_path(col_item, "location")
+      field "kind", jmes_path(col_item, "kind")
+      field "properties", jmes_path(col_item, "properties")
+      field "tags", jmes_path(col_item, "tags")
+      field "subscriptionId", val(iter_item, "subscriptionId")
+      field "subscriptionName", val(iter_item, "subscriptionName")
+    end
+  end
+end
+
+datasource "ds_sql_servers_firewall_settings" do
+  iterate $ds_sql_servers
+  request do
+    auth $azure_auth
+    pagination $azure_pagination
+    host $param_azure_endpoint
+    path join([val(iter_item, 'id'), "/firewallRules"])
+    query "api-version", "2021-02-01-preview"
+    header "User-Agent", "RS Policies"
+  end
+
+
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "id", val(iter_item, "id")
+      field "name", val(iter_item, "name")
+      field "startIpAddress", jmes_path(col_item, "properties.startIpAddress")
+      field "endIpAddress", jmes_path(col_item, "properties.endIpAddress")
+      field "type", jmes_path(col_item, "type")
+    end
+  end
+end
+
+datasource "ds_bad_sql_servers" do
+  run_script $js_bad_sql_servers, $ds_sql_servers_firewall_settings
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+script "js_bad_sql_servers", type: "javascript" do
+  parameters "ds_sql_servers_firewall_settings"
+  result "result"
+  code <<-EOS
+  result = []
+
+  _.each(ds_sql_servers_firewall_settings, function(server) {
+    if (server['startIpAddress'] == '0.0.0.0') {
+      result.push(server)
+    }
+  })
+EOS
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "policy_azure_resource" do
+  validate $ds_bad_sql_servers do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): Azure SQL Databases with ingress startingIpAddress set to 0.0.0.0/0"
+    escalate $escalate_email
+    check eq(size(data),0)
+    export do
+      resource_level true
+      field "id" do
+        label "Azure Resource Id"
+      end
+      field "name" do
+        label "SQL Database Name"
+      end
+      field "startIpAddress" do
+        label "Starting Range IP Address"
+      end
+      field "endIpAddress" do
+        label "Ending Range IP Address"
+      end
+    end
+  end
+end
+
+###############################################################################
+# Escalations
+###############################################################################
+
+escalation "escalate_email" do
```

Comment:
should use "_esc" prefix for escalations as opposed to "_escalate"- see https://confluence.flexera.com/pages/viewpage.action?spaceKey=rightscale&title=Policy+Template+Development+Best+Practices